pkg/login/ext_user.go

```diff
@@ -0,0 +1,171 @@
+package login
+
+import (
+	"fmt"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/log"
+	m "github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/quota"
+)
+
+var UpsertUser = func(ctx *m.ReqContext, cmd *m.UpsertUserCommand) error {
```

Comment:
I'm torn about this. I don't like exposing functions as variables globally within Grafana. Internally in packages, its fine but globally feels dirty since anyone can modify. 

I'm leaning towards `DispatchReqCtx` but then we have three functions for dispatching... It would be possible to combine `Dispatch` and `DispatchCtx` since its possible to just send `context.Background()` when you don't have one. but... it feels hacky. Have to think about this. This is not a blocker btw. Just me mumbling.

Reply:
I guess the other option would be to have the query/command carry the ReqContext, wasn't sure if that was a good idea though